sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/ParentElement.java

```diff
@@ -100,47 +98,48 @@ default Status getStatus() {
         // Ordering matters throughout this method.  Modify with care.
         // Also note that this function MUST NOT call parent.getStatus() as that creates a circular call.
 
-        final Collection<? extends Element> children = getChildren();
+        final List<C> children = getChildren();
         if (children == null) {
             LOGGER.error("Parent element returned null list of children: {}", getName());
             return Status.ERROR;
         }
 
+        final Collection<? extends Element> candidateChildren =
```

Comment:
Parent Status depends on its strategy.  A parent with "PENDING, WAITING" children should be PENDING, whereas a parent with "WAITING, PENDING" children should be WAITING.